frontend/packages/console-app/src/hooks/useCSPVioliationDetector.tsx

```diff
@@ -154,49 +155,47 @@ const useCSPViolationReporter: UseCSPVilationReporter = () => {
         }
       }
     },
-    [fireTelemetryEvent, getRecords, t, toastContext, updateRecords],
-  );
-};
-
-export const useCSPViolationDetector = () => {
-  const reportViolation = useCSPViolationReporter();
-  const onCSPViolation = React.useCallback(
-    (event) => {
-      // Attempt to infer Console plugin name from SecurityPolicyViolation event
-      const pluginName =
-        getPluginNameFromResourceURL(event.blockedURI) ||
-        getPluginNameFromResourceURL(event.sourceFile);
-
-      reportViolation(pluginName, event);
-    },
-    [reportViolation],
+    [fireTelemetryEvent, getRecords, t, toastContext, updateRecords, pluginStore],
   );
 
   React.useEffect(() => {
-    document.addEventListener('securitypolicyviolation', onCSPViolation);
+    document.addEventListener('securitypolicyviolation', reportViolation);
     return () => {
-      document.removeEventListener('securitypolicyviolation', onCSPViolation);
+      document.removeEventListener('securitypolicyviolation', reportViolation);
     };
-  }, [onCSPViolation]);
+  }, [reportViolation]);
 };
 
+// A subset of properties from a SecurityPolicyViolationEvent which identify a unique CSP violation
 type CSPViolationReportProperties =
+  // The URI of the resource that was blocked because it violates a policy.
   | 'blockedURI'
+  // The column number in the document or worker at which the violation occurred.
   | 'columnNumber'
+  // Whether the user agent is configured to enforce or just report the policy violation.
   | 'disposition'
+  // The URI of the document or worker in which the violation occurred.
   | 'documentURI'
+  // The directive that was violated.
   | 'effectiveDirective'
+  // The line number in the document or worker at which the violation occurred.
   | 'lineNumber'
+  // The policy whose enforcement caused the violation.
   | 'originalPolicy'
+  // The URL for the referrer of the resources whose policy was violated, or null.
   | 'referrer'
+  // A sample of the resource that caused the violation, usually the first 40 characters.
+  // This will only be populated if the resource is an inline script, event handler or style.
   | 'sample'
+  // If the violation occurred as a result of a script, this will be the URL of the script.
   | 'sourceFile'
+  // HTTP status code of the document or worker in which the violation occurred.
   | 'statusCode';
+
+// A CSPViolationReport represents a unique CSP violation per plugin
 type CSPViolationReport = Pick<SecurityPolicyViolationEvent, CSPViolationReportProperties> & {
   pluginName: string;
 };
+
+// A CSPViolationRecord represents a unique CSP violation per plugin, per occurrance
```

Comment:
just a nitpick but this should use jsdoc format for better Intellisense integration